### PR TITLE
Do not error out if no ip is available in a prefix

### PR DESCRIPTION
**Pull request by majst01**
closes: #550 
Depends on: https://github.com/metal-stack/go-ipam/pull/157

### CODE REVIEW

**Review by Gerrit91 — APPROVED** [review]

File: cmd/metal-api/internal/service/ip-service.go [hunk 1]
```diff
@@ -439,8 +437,11 @@ func allocateSpecificIP(ctx context.Context, parent *metal.Network, specificIP s
 func allocateRandomIP(ctx context.Context, parent *metal.Network, ipamer ipam.IPAMer) (ipAddress, parentPrefixCidr string, err error) {
 	for _, prefix := range parent.Prefixes {
 		ipAddress, err = ipamer.AllocateIP(ctx, prefix)
-		if err != nil && errors.Is(err, goipam.ErrNoIPAvailable) {
```
[the remark this answers is not in the document]
Comment (majst01) [reply cmd/metal-api/internal/service/ip-service.go:442]: Now

File: cmd/metal-api/internal/service/network-service.go [hunk 2]
```diff
@@ -560,6 +561,12 @@ func (r *networkResource) freeNetwork(request *restful.Request, response *restfu
 	for _, prefix := range nw.Prefixes {
 		err = r.ipamer.ReleaseChildPrefix(ctx, prefix)
 		if err != nil {
+			var connectErr *connect.Error
+			if errors.As(err, &connectErr) {
```
Comment (Gerrit91) [inline cmd/metal-api/internal/service/network-service.go:565]: Probably we cannot skip here as then we have orphaned prefixes after freeing the network.

File: cmd/metal-api/internal/service/ip-service.go [hunk 3]
```diff
@@ -439,8 +437,11 @@ func allocateSpecificIP(ctx context.Context, parent *metal.Network, specificIP s
...
-			continue
+		var connectErr *connect.Error
```
Comment (majst01) [inline cmd/metal-api/internal/service/ip-service.go:440]: check for err != nil is missing here

File: cmd/metal-api/internal/service/network-service.go [hunk 4]
```diff
@@ -730,6 +743,12 @@ func (r *networkResource) deleteNetwork(request *restful.Request, response *rest
 	for _, p := range nw.Prefixes {
 		err := r.ipamer.DeletePrefix(ctx, p)
 		if err != nil {
+			var connectErr *connect.Error
```
Comment (Gerrit91) [inline cmd/metal-api/internal/service/network-service.go:746]: Same here.

**Pull request merged**